portfolio/src/main/java/com/google/sps/servlets/BusinessCardServlet.java

```diff
@@ -0,0 +1,173 @@
+    response.sendRedirect("/index.html");
+  }
+
+  /**
+   * Return a list of blobKeys for all business cards in datastore.
+   * @param request the HTTP request sent from client for GET
+   * @param response HTTP response that will be sent back to the client
+   * @throws IOException if an IO error occurs while the request is being processed by the servlet.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Create Query to get the blobKeys
+    Query query = new Query("BizCard");
+
+    // Get the Entities and populate the array list with the blobKeys
+    List<Entity> results = datastore.prepare(query).asList(FetchOptions.Builder.withDefaults());
+    List<String> blobKeys = results.stream().map(entity -> (String) entity.getProperty("bizCard"))
+        .collect(Collectors.toList());
+
+    // Convert list to JSON
+    String blobKeyJson = HttpServletUtilities.listToJson(blobKeys);
+
+    // Attach the JSON to the response
+    response.setContentType("application/json;");
+    response.getWriter().println(blobKeyJson);
+  }
+
+  /**
+   * This function will get the blobkey of an uploaded image from blobstore.
+   * Note this will only get the blobkey of the first image in the upload
+   * (If more than one file is uploaded per form element, this function will return
+   * unexpected results).
+   * @param request The HTTP request sent by blobstore
+   * @param paramName the name of the parameter in the request that stores the image
+   * @return blobkey of image
+   * @throws NullPointerException if no file was uploaded
+   * @throws IllegalArgumentException if the file was not an image
+   */
+  private BlobKey getImageBlobKey(HttpServletRequest request, String paramName)
+      throws FileNotFoundException, IllegalArgumentException {
+    // Get the relevant blobkeys from the blobstore
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(paramName);
+
+    // Make sure at least one blob was uploaded
+    // In Dev, an empty file will be a null blob
+    // (thus this error will trigger in dev)
+    if (blobKeys == null || blobKeys.size() == 0) {
+      throw new FileNotFoundException();
+    }
+
+    // Get first image in the blob keys list
+    // Change this if more than one blobKey is to be returned.
+    BlobKey blobKey = blobKeys.get(0);
+
+    // Check the image file
+    // Make sure the file isn't empty and is of type image
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+
+    // In prod, an empty file will be non-null but empty.
+    // (thus this error will trigger in prod)
+    if (blobInfo.getSize() == 0) {
+      blobstoreService.delete(blobKey);
+      throw new FileNotFoundException();
+    }
+
+    if (!blobInfo.getContentType().contains("image")) {
+      blobstoreService.delete(blobKey);
+      throw new IllegalArgumentException();
```

Comment:
Nit: It could be useful for debugging to add to the exception as a message what the content type was

Reply:
Yes, I agree.

Reply:
Good idea - changed!